### PR TITLE
Tooltips

### PR DESCRIPTION
**Pull request by jgozal**
Looking to get feedback, not just on the code, but also on the text content and the aesthetics/styling. Please pull and give it a try

image

### CODE REVIEW

File: src/components/general/validated-number-input.js [hunk 1]
```diff
@@ -40,6 +40,8 @@ export default function (props) {
     <Input
       invalid={!isValid}
       name={props.name}
+      min={props.min}
+      max={props.max}
```
Comment (jgozal) [inline src/components/general/validated-number-input.js]: this works, trust me. I'll show you in screenshare some time

File: src/components/mortgage-investment-compare/input-card.js [hunk 2]
```diff
@@ -32,13 +34,18 @@ const InputCard = props => {
...
+            content='The amount you promise to pay back after making your down payment'
+            id='loanAmt'
+          />
           <InputGroup className='my-2'>
             <InputGroupAddon addonType='prepend'>
               <InputGroupText>$</InputGroupText>
             </InputGroupAddon>
             <ValidatedNumberInput
-              max={500000000}
+              max={5000000000}
```
Comment (jgozal) [inline src/components/mortgage-investment-compare/input-card.js]: I really wanted to buy the Buckingham Palace. No issues from having an extra 0, our app handles it gracefully

File: src/components/mortgage-investment-compare/input-card.js [hunk 3]
```diff
@@ -123,16 +135,21 @@ const InputCard = props => {
...
             <Col>
               <small>
                 <b>Mortgage Option 1</b>
               </small>
               <InputGroup className='my-2'>
                 <ValidatedNumberInput
-                  max={100}
-                  min={-100}
+                  max={1000}
+                  min={-1000}
```
Comment (jgozal) [inline src/components/mortgage-investment-compare/input-card.js]: added an extra 0 to handle predatory ~200-500% loans

File: src/components/mortgage-investment-compare/input-card.js [hunk 4]
```diff
@@ -177,11 +194,16 @@ const InputCard = props => {
...
+            content='How well you expect your investment to perform each year'
+            id='investmentRate'
+          />
           <InputGroup className='my-2'>
             <ValidatedNumberInput
-              max={100}
-              min={-100}
+              max={1000}
+              min={-1000}
               name='investmentRate'
```
Comment (jgozal) [inline src/components/mortgage-investment-compare/input-card.js]: bitcoin

File: src/components/mortgage-investment-compare/input-card.js [hunk 5]
```diff
@@ -250,11 +272,16 @@ const InputCard = props => {
...
+            icon={faQuestionCircle}
+            content='How fast you expect prices to rise every year'
+            id='inflation'
+          />
           <InputGroup className='my-2'>
             <ValidatedNumberInput
-              max={200}
-              min={-200}
+              max={1000000000}
+              min={-1000000000}
```
Comment (jgozal) [inline src/components/mortgage-investment-compare/input-card.js]: yes, this is possible

File: src/index.js [hunk 6]
```diff
@@ -10,7 +10,7 @@ import ReactDOM from 'react-dom'
 import { BrowserRouter as Router, Route } from 'react-router-dom'
 
 import { GRAY_900, BLUE_GRAY_900, LIGHT_BLUE } from './assets/colors'
-import FeedbackButton from './components/feedback-button'
+import FeedbackButton from './components/general/feedback-button'
```
Comment (jgozal) [inline src/index.js]: I moved the general stuff to the new general folder

File: src/components/mortgage-investment-compare/index.js [hunk 7]
```diff
@@ -35,7 +35,7 @@ class MortgageInvestmentCompare extends React.Component {
 
     const input = {
       inflation: 2,
-      investmentRate: 9,
+      investmentRate: 8,
```
Comment (jgozal) [inline src/components/mortgage-investment-compare/index.js]: it's actually 8%

File: src/components/mortgage-investment-compare/input-card.js (hunk 5 above)
Comment (hdavidzhu) [reply src/components/mortgage-investment-compare/input-card.js]: An issue I ran into is when you max out all the numbers, then the app crashes. It doesn't happen when you max out an individual field

File: src/components/general/tooltip.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { Tooltip } from 'reactstrap'
+import { css } from 'react-emotion'
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+export default class TooltipWrapper extends React.Component {
```
Comment (hdavidzhu) [inline src/components/general/tooltip.js]: Should this have a different name? The file's called `tooltip`

File: src/components/mortgage-investment-compare/input-card.js (hunk 5 above)
Comment (jgozal) [reply src/components/mortgage-investment-compare/input-card.js]: maybe 1.7976931348623157e+308? lol I'll give it a try later

File: src/components/mortgage-investment-compare/input-card.js [hunk 9]
```diff
@@ -123,16 +135,21 @@ const InputCard = props => {
         </FormGroup>
         <hr />
         <FormGroup className='mb-3'>
-          <label>Annual Percentage Rate (APR) & Interest</label>
+          <Tooltip
+            children={<label>Annual Percentage Rate (APR) & Interest</label>}
```
Comment (hdavidzhu) [inline src/components/mortgage-investment-compare/input-card.js]: sug: How about

```html
<Tooltip>
  <label/>
</Tooltip>
```

to better match HTML structure?

File: src/components/mortgage-investment-compare/input-card.js [hunk 10]
```diff
@@ -206,7 +228,7 @@ const InputCard = props => {
           <Accordion title='Return on Investment (ROI)'>
             Return on investment (ROI) is the gain or loss generated on your
             investment relative to the amount of money invested (expressed as a
-            percentage). This tool defaults to a 9% ROI because that is roughly
+            percentage). This tool defaults to a {props.input.investmentRate}% ROI because that is roughly
```
Comment (hdavidzhu) [inline src/components/mortgage-investment-compare/input-card.js]: Wouldn't this be incorrect? If the user changes the investment rate, the copy would say that this is what we are defaulting to

Comment (jgozal) [reply src/components/mortgage-investment-compare/input-card.js]: you're right. Good catch. will correct

File: src/components/mortgage-investment-compare/input-card.js (hunk 9 above)
Comment (jgozal) [reply src/components/mortgage-investment-compare/input-card.js]: yeah that's better, ill make the change

File: src/components/general/tooltip.js (hunk 8 above)
Comment (jgozal) [reply src/components/general/tooltip.js]: any suggestions? We already have <Tooltip> being imported from reactsrap :/

**Pull request merged**